### PR TITLE
feat(HW-884): Cross-chain swaps - toAddress screen

### PR DESCRIPTION
**Pull request by devkudasov**
## Proposed Changes

- [Add network select to Address screen](https://linear.app/haqq/issue/HW-885/add-network-select-to-address-screen)
- [Network auto select](https://linear.app/haqq/issue/HW-886/network-auto-select)
- [Disable networks](https://linear.app/haqq/issue/HW-887/disable-networks)
- [Select network screen re-design](https://linear.app/haqq/issue/HW-888/select-network-screen-re-design)

## Checklist

- [X] I have read the rules of [**contribution**](https://github.com/haqq-network/haqq-wallet/blob/main/docs/contribution.md#important-steps).

### CODE REVIEW

File: src/screens/HomeStack/TransactionStack/transaction-address/transaction-address-wallet-list.tsx [hunk 1]
```diff
@@ -1,48 +1,91 @@
...
+  ({onPress}: TransactionAddressWalletListProps) => {
+    const {toAddress, fromAddress} = TransactionStore;
+
+    const filteredWallets = useMemo(() => {
+      const wallets = Wallet.getAllVisible();
+
+      if (!wallets?.length) {
+        return [];
+      }
+      const isTron = Provider.selectedProvider.isTron;
```
Comment (iGroza) [inline src/screens/HomeStack/TransactionStack/transaction-address/transaction-address-wallet-list.tsx:28]: get provider from TransactionStore

File: src/screens/HomeStack/TransactionStack/transaction-address/transaction-address.tsx [hunk 2]
```diff
@@ -26,109 +29,39 @@ import {NetworkProviderTypes} from '@app/services/backend';
...
-      // if (contacts?.length === 1 && filteredWallets?.length === 0) {
-      //   return onDone(contacts[0].account);
-      // }
-
-      // if (AddressUtils.isValidAddress(address.trim())) {
-      //   return onDone(address.trim());
-      // }
-
       try {
         const networkType = Provider.selectedProvider.isTron
```
Comment (iGroza) [inline src/screens/HomeStack/TransactionStack/transaction-address/transaction-address.tsx:66]: refactor Provider.selectedProvider to TransactionStore

**Review by iGroza — APPROVED** [review]

File: src/screens/HomeStack/TransactionStack/transaction-address/transaction-address-wallet-list.tsx (hunk 1 above)
Comment (devkudasov) [reply src/screens/HomeStack/TransactionStack/transaction-address/transaction-address-wallet-list.tsx]: Fixed

File: src/screens/HomeStack/TransactionStack/transaction-address/transaction-address.tsx (hunk 2 above)
Comment (devkudasov) [reply src/screens/HomeStack/TransactionStack/transaction-address/transaction-address.tsx]: Done

**Review by iGroza — APPROVED** [review]

**Comment by devkudasov** [comment]
To many dependencies between new screens. All changes for all cross chain tasks will be completed into [this PR](https://github.com/haqq-network/haqq-wallet/pull/2237)

**Pull request closed without merging**